pkg/table/tables/tables.go

```diff
@@ -260,6 +260,11 @@ func initTableCommon(t *TableCommon, tblInfo *model.TableInfo, physicalTableID i
 	t.writableColumnsOnce = new(sync.Once)
 	t.fullHiddenColsAndVisibleColumnsOnce = new(sync.Once)
 	t.writableConstraintsOnce = new(sync.Once)
+	_ = t.Cols()
+	_ = t.VisibleCols()
+	_ = t.HiddenCols()
+	_ = t.WritableCols()
+	_ = t.FullHiddenColsAndVisibleCols()
```

Comment:
This PR seems to go the opposite way from https://github.com/pingcap/tidb/pull/50062

Reply:
Now we call `Cols` using `t.publicColumnsOnce`. It feels like there should be no DataRace. Could you explain it here? @tiancaiamao 
https://github.com/pingcap/tidb/pull/53798/files#diff-2b615fee1d4e99193f0469ddef5bb5fa6d372da03767dc4643c24e4982424a1dR354

Reply:
This data race comes from infoschema V2, and can't be get rid of by using sync.Once.
In infoschema v2, we have cache for tables instead of loading all tables into memory.
Before a table is put into cache, its size need to be calculated. 
How to calculate the size? we use reflect to visit the field of an object recursively...
So when calling sizeof, we're not using the Cols() API, but we do visit the fields of table's columns.
This is the reader part of the DATA RACE.

When someone use the table object, and the lazy initialization logic would modify the object.
So the DATA RACE happen. @zimulala

Reply:
only do it for v2?

Reply:
Currently, I think we can consider dealing with v2 only.
Or if we really want to do it like this PR, how about getting rid of sync.Once?

Reply:
Done